### PR TITLE
Add new annotation type RotatedBbox

### PR DESCRIPTION
**Pull request by wonjuleee**
### Summary

Add a new RotatedBbox annotation type for detection_rotated task.
This provides from_polygon and as_polygon methods for converting set of coordinates to (cx, cy, w, h, r).

This updates the RoboflowYoloObb behavior as a detection_rotated task.

### Checklist

- [x] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: tests/unit/test_annotation.py [hunk 1]
```diff
@@ -52,3 +52,18 @@ def fxt_hashkeys_diff(self):
...
+
+
+class RotatedBboxTest:
+    @pytest.fixture
+    def fxt_rot_bbox(self):
+        coords = np.random.randint(0, 180, size=(5,), dtype=np.uint8)
+        return RotatedBbox(coords[0], coords[1], coords[2], coords[3], coords[4])
+
+    @pytest.mark.parametrize("fxt_ann", ["fxt_rot_bbox"])
+    def test_create_polygon(self, fxt_ann, request):
```
Comment (itrushkin) [inline tests/unit/test_annotation.py]:
```suggestion
    def test_create_polygon(self, fxt_rot_bbox):
```
You can pass the name of the fixture as a parameter. `parametrize` decorator can also be removed.

File: src/datumaro/components/annotation.py [hunk 2]
```diff
@@ -845,6 +847,101 @@ def wrap(item, **kwargs):
...
+@attrs(slots=True, init=False, order=False)
+class RotatedBbox(_Shape):
+    _type = AnnotationType.rotated_bbox
+
+    def __init__(self, cx, cy, w, h, r, *args, **kwargs):
+        kwargs.pop("points", None)  # comes from wrap()
+        self.__attrs_init__([cx, cy, w, h, r], *args, **kwargs)
+
+    @classmethod
+    def from_polygon(cls, points: List[Tuple[float, float]], *args, **kwargs):
```
Comment (itrushkin) [inline src/datumaro/components/annotation.py]: This function accepts only rectangle-like polygon. `from_rectangle` name could be clearer. What do you think?

File: tests/unit/data_formats/test_roboflow.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 import numpy as np
 import pytest
 
-from datumaro.components.annotation import Bbox, Label, Polygon
+from datumaro.components.annotation import Bbox, Label, Polygon, RotatedBbox
```
Comment (itrushkin) [inline tests/unit/data_formats/test_roboflow.py]: Please remove `Polygon` import as it is not accessed anymore

File: tests/unit/test_annotation.py (hunk 1 above)
Comment (wonjuleee) [reply tests/unit/test_annotation.py]: fixed at 2bbfbaeee81b4c8f89a67ccc937dce9c7c7d4aef

File: src/datumaro/components/annotation.py (hunk 2 above)
Comment (wonjuleee) [reply src/datumaro/components/annotation.py]: fixed at 2bbfbaeee81b4c8f89a67ccc937dce9c7c7d4aef

File: tests/unit/data_formats/test_roboflow.py (hunk 3 above)
Comment (wonjuleee) [reply tests/unit/data_formats/test_roboflow.py]: fixed at 2bbfbaeee81b4c8f89a67ccc937dce9c7c7d4aef

**Review by itrushkin — APPROVED** [review]

**Pull request merged**